File: frontend/src/types/game.ts
export type Suit = 'hearts' | 'diamonds' | 'clubs' | 'spades';
export type Rank = number; // 2-14 (2-10, J=11, Q=12, K=13, A=14)

export interface Card {
  rank: Rank;
  suit: Suit;
  display?: string; // e.g. "A♥" — provided by backend
}

export type PlayerStatus = 'active' | 'folded' | 'all_in' | 'sitting_out';
export type GamePhase = 'pre_flop' | 'flop' | 'turn' | 'river' | 'showdown' | 'waiting' | 'finished';

export interface Player {
  id: string;
  name: string;
  chips: number;
  cards: Card[];
  status: PlayerStatus;
  current_bet: number;
  is_dealer: boolean;
  is_small_blind: boolean;
  is_big_blind: boolean;
}

export interface GameState {
  game_id: string;
  table_id: string;
  phase: GamePhase;
  players: Record<string, Player>;
  community_cards: Card[];
  pot: number;
  current_bet: number;
  current_player_id: string | null;
  hand_number: number;
}

export interface TableInfo {
  id: string;
  name: string;
  player_count: number;
  max_players: number;
  small_blind: number;
  big_blind: number;
  status: 'waiting' | 'playing' | 'finished';
  game_id?: string;
}

export interface AgentInfo {
  id: string;
  name: string;
  type: string;
  created_at?: string;
}

export interface PlatformStats {
  total_games: number;
  active_tables: number;
  total_agents: number;
  total_hands: number;
}

export interface ActionEvent {
  player: Player;
  action: string;
  amount: number | null;
  pot: number;
}

export interface PhaseChangeEvent {
  phase: GamePhase;
  community_cards: Card[];
}

export interface HandResultEvent {
  winners: Array<{ player: Player; amount: number; hand_description?: string }>;
  pot: number;
}

export interface ActionLogEntry {
  id: string;
  timestamp: Date;
  type: 'action' | 'phase_change' | 'hand_result' | 'player_joined' | 'error' | 'info';
  message: string;
  data?: unknown;
}

File: backend/app/schemas/game.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel


class TableCreate(BaseModel):
    name: str
    max_players: int = 6
    min_players: int = 2
    small_blind: int = 10
    big_blind: int = 20
    starting_chips: int = 1000


class TableResponse(BaseModel):
    id: str
    name: str
    max_players: int
    min_players: int
    small_blind: int
    big_blind: int
    starting_chips: int
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class TableDetailResponse(TableResponse):
    players: List[Dict[str, Any]] = []


class GameActionRequest(BaseModel):
    action: str
    amount: Optional[int] = None


class GameResponse(BaseModel):
    id: str
    table_id: str
    hand_number: int
    phase: str
    status: str
    pot: int
    community_cards: List[Any]
    started_at: datetime
    finished_at: Optional[datetime] = None
    winner_info: Optional[Any] = None

    model_config = {"from_attributes": True}


class ValidActionsResponse(BaseModel):
    player_id: str
    actions: List[Dict[str, Any]]


class StatsResponse(BaseModel):
    total_agents: int
    total_tables: int
    total_games: int
    active_games: int
